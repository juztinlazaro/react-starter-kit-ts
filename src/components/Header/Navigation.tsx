import React from 'react';
import { Link } from 'react-router-dom';
import { useAppContext } from 'appRoot/App.context';

const Navigation = () => {
  const { isLoggedIn, onLogin, onLogout } = useAppContext();

  return (
    <div className="item link-items">
      {isLoggedIn ? (
        <>
          <Link to="/" className="link">
            Home
          </Link>

          <Link to="/test" className="link">
            Test
          </Link>

          <Link to="/public" className="link">
            Public
          </Link>

          <button className="action-button logout" onClick={onLogout}>
            Logout
          </button>
        </>
      ) : (
        <button className="action-button login" onClick={onLogin}>
          Login
        </button>
      )}
    </div>
  );
};

export default Navigation;
