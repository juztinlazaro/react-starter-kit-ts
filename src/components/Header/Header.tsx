import React from "react";
import ReactLogo from "components/ReactLogo/ReactLogo";
import Navigation from "./Navigation";

const Header = () => {
  return (
    <section className="header-section">
      <div className="item">
        <span className="title">
          <ReactLogo />
        </span>
      </div>

      <Navigation />
    </section>
  );
};

export default Header;
