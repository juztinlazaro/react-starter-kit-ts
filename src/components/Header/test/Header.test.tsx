import React from "react";
import { shallow } from "enzyme";
import Header from "../Header";

describe("Header test component", () => {
  const HeaderWrapper = shallow(<Header />);

  it("should match to snapshot", () => {
    expect(HeaderWrapper).toMatchSnapshot();
  });
});
