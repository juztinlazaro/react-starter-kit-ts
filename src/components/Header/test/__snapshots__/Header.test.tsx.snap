// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test component should match to snapshot 1`] = `ShallowWrapper {}`;
